CK3toEU4/Source/EU4World/FlagFoundry/FlagFoundry.cpp

```diff
@@ -97,20 +103,56 @@ void EU4::FlagFoundry::craftFlag(const std::shared_ptr<Country>& country) const
 	}
 }
 
+void EU4::FlagFoundry::craftRebelFlag(const Configuration& theConfiguration, const GeneratedReligion& religion) const
+{
+	//Import the generic Rebel Flag	
+	if (!commonItems::DoesFileExist("blankMod/output/gfx/flags/generic_rebels.tga"))
+		throw std::runtime_error("blankMod/output/gfx/flags/generic_rebels.tga! Where are the rebel scum!?");
+	Magick::Image baseFlag("blankMod/output/gfx/flags/generic_rebels.tga");
+	
+	//Process target icon
+	Magick::Image targetIcon;
+	if (!religion.iconPath.empty() && commonItems::DoesFileExist(theConfiguration.getCK3Path() + "gfx/interface/icons/religion/" + religion.iconPath + ".dds"))
```

Comment:
Yes and no. the you gave icon="catholic" this works, when you have icon="gfx/interface/icons/religion/catholic.dds" (as it happens for custom religions, it wont.
You need to expand the search for icon, see line 107 of updated FlagFoundry.cpp. 
Sync first!